packages/async-rewriter/src/async-writer-babel.spec.ts

```diff
@@ -3137,4 +3137,65 @@ switch(TEST) {
       });
     });
   });
+  describe('Assign API type', () => {
+    before(() => {
+      writer = new AsyncWriter(signatures);
+      writer.symbols.initializeApiObjects({ db: signatures.Database });
+    });
+    it('init', () => {
+      expect(writer.symbols.scopeAt(0).db).to.deep.equal({
+        api: true, ...signatures.Database
+      });
+    });
+    it('regular add', (done) => {
+      input = 'const db = 1';
+      try {
+        writer.compile(input);
+      } catch (err) {
+        done();
```

Comment:
if is sync just removing done and the try catch would do the same:

```
it('regular add', (done) => {
      input = 'const db = 1';
      writer.compile(input);
 });
```

Reply:
But if the error isn't thrown then it wouldn't notice if there's no done() statement, right?  It's basically mocha expect(...).to.throw() except not buggy

Reply:
Sorry mistyped the example:

```
it('regular add', () => {
      input = 'const db = 1';
      writer.compile(input);
 });
```

This would fail if  anything inside throws and succeed otherwise.

`done` is only needed if you are testing something that is really asynchronous and has a callback, for example:

```
function returnAfterTimeout(ms, callback) {
    setTimeout(() => callback(null), ms);
}

it('returns at some point', (done) => {
   returnAfterTimeout(done);
});
```

Reply:
But we want the error to be thrown, since there's no "try/catch/else" statement and mocha's expect.to.be.thrown is buggy. How would you ensure that an error is thrown without using done() and without using expect.to.be.thrown?

Reply:
Perhaps the example will make more sense now that I've updated it with the custom errors that were just merged.

Reply:
Ohh now i got it, why `expect.to.be.thrown` is buggy?